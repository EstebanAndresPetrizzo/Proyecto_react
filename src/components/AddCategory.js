import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {
    const [inputValue, setInputValue] = useState('');
    const handleinputChange = (e)=>{
        setInputValue(e.target.value);
    }

    const handleSubmit = (e)=>{
        e.preventDefault();//Evita que se refresque la pagina
        if(inputValue.trim().length > 2){
            setCategories(cats => [inputValue,...cats]);
        }
    }
    return (
        <form onSubmit= {handleSubmit}>
            <p>{inputValue}</p>
            <input type="text"
            value={inputValue}
            onChange={handleinputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
