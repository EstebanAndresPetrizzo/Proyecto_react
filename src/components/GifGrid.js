import React/*, { useState, useEffect } */from 'react'
import PropTypes from 'prop-types'

import {useFechGifs} from '../hooks/useFechGifs'
import { GifGridItem } from './GifGridItem';
//import { getGifs } from './helpers/getGifs';

export const GifGrid = ({ category }) => {

    const {data:images,loading} = useFechGifs(category)
/*
    const [images, setImages] = useState([]);
    useEffect(() => {
        getGifs(category)
            .then(setImages);
    }, [category])//el segundo argumento son las dependencias, si se deja vac√≠o [], entonces se ejecutara una sola vez
*/
    return (
        <>
            <h3 className="animate__animated animate__zoomIn">{category}</h3>
            {loading && <p className="animate__animated animate__flash">Loading...</p>}
            <div className="card-grid">
                {
                    images.map(img => (
                        <GifGridItem
                            key={img.id}
                            {...img}//envia cada una de la propiedades
                        />
                    ))
                }
            </div>
        </>
    )
}

GifGrid.protoType = {
    category: PropTypes.string.isRequired
}