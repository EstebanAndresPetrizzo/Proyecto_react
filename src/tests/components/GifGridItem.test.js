import React from 'react';
import {shallow} from 'enzyme';
import {GifGridItem} from '../../components/GifGridItem';

describe('Prueba en <GifGridItem />', () => {
  
    let wrapper = shallow(<GifGridItem />);
    beforeEach(() => {
        wrapper = shallow(<GifGridItem />);
    });

    test('Debe mostrar <GifGridItem /> correctamente', () =>{
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe tener animate__zoomIn', () => {
        const div = wrapper.find('div');
        const className = div.prop('className');
        expect(className.includes('animate__zoomIn')).toBe(true);
    });
    
});