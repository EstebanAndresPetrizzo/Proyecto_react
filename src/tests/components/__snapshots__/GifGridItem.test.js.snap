// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <GifGridItem /> Debe mostrar <GifGridItem /> correctamente 1`] = `
<div
  className="animate__animated animate__zoomIn"
>
  <img />
  <p />
</div>
`;
